tests/integration/components/sl-calendar-test.js

```diff
@@ -0,0 +1,773 @@
+            assert.equal(
+                dateContent[ 0 ].date.toString(),
+                'Tue Sep 20 2022 00:00:00 GMT-0500 (CDT)',
+                'The date property was passed through'
+            );
+
+            assert.equal(
+                dateContent[ 0 ].label,
+                'Event 1 Another Day!',
+                'The label property was passed through'
+            );
+        }
+    });
+
+    this.$( '.sl-calendar .active' ).click();
+});
+
+test( 'Setting viewMode modifies the view correctly', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            viewMode="months"
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        'Sep',
+        'The current month is set correctly'
+    );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            viewMode="years"
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        '2020-2029',
+        'The year range is set correctly'
+    );
+
+    assert.equal(
+        this.$( '.sl-calendar .active' ).text().trim(),
+        currentYear,
+        'The current year is set correctly'
+    );
+});
+
+test( 'Navigating Forward by Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
+
+    this.render( hbs`
+        {{sl-calendar
+            action='testAction'
+            content=content
+            currentYear=currentYear
+            currentMonth=currentMonth
+        }}
+    ` );
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'September ' + currentYear,
+        'The current month is set correctly'
+    );
+
+    this.$( '.sl-calendar .next' ).click();
+
+    assert.equal(
+        this.$( '.sl-calendar .datepicker-switch' ).text().trim(),
+        'October ' + currentYear,
+        'The next month is set correctly'
+    );
+});
+
+test( 'Navigating Backward by Month', function( assert ) {
+
+    const currentYear = 2022;
+    this.set( 'currentYear', currentYear );
+
+    const currentMonth = 9;
+    this.set( 'currentMonth', currentMonth );
+
+    this.set( 'content', testContent );
```

Comment:
What role does this serve in this test? The component will still render as expected, especially within the context of this test, without this.